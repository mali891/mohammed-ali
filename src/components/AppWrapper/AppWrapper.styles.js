import { css } from '@emotion/react'
import { COLOURS, EASINGS } from '../../styles/settings'

export const styles = css`
  transition: ${EASINGS.DEFAULT};
  background-color: var(--colour-primary);
  color: var(--colour-secondary);

  &.theme--light {
    --colour-primary: ${COLOURS.LIGHT};
    --colour-secondary: ${COLOURS.DARK};
    --colour-tertiary: ${COLOURS.GREY};
  }
`
