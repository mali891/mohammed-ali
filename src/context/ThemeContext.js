import React, { createContext, useState } from 'react'
import PropTypes from 'prop-types'

const { Provider, Consumer } = createContext()

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('dark')

  return <Provider value={{ theme, setTheme }}>{children}</Provider>
}

ThemeProvider.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
}

export { ThemeProvider, Consumer as ThemeConsumer }
