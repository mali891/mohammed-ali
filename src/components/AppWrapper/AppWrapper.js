import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { HeadHelmet } from '../'
import { GlobalStyles } from '../../styles/settings'
import { ThemeConsumer } from '../../context/ThemeContext'
import { styles } from './AppWrapper.styles'

const AppWrapper = ({ children }) => {
  const dynamicClassNames = classNames([], {
    'c-app-wrapper': true,
    'u-anim--fade-in': true
  })

  return (
    <ThemeConsumer>
      {({ theme }) => (
        <div css={styles} className={`${dynamicClassNames} theme--${theme}`}>
          <GlobalStyles />
          <HeadHelmet />
          {children}
        </div>
      )}
    </ThemeConsumer>
  )
}

AppWrapper.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
}

export default AppWrapper
