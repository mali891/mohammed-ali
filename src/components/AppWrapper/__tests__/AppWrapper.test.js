import React from 'react'
import { render } from '@testing-library/react'

import AppWrapper from '../AppWrapper'
import { ThemeProvider } from '../../../context/ThemeContext'

describe('AppWrapper', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <AppWrapper>
          <p>Dummy children</p>
        </AppWrapper>
      </ThemeProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
