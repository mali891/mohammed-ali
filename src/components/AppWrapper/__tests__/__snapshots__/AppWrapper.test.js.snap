// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppWrapper renders correctly 1`] = `
<DocumentFragment>
  <div
    class="c-app-wrapper u-anim--fade-in theme--dark"
    css="[object Object]"
  >
    <p>
      Dummy children
    </p>
  </div>
</DocumentFragment>
`;
